sound/soc/intel/boards/sof_sdw_rt711.c

```diff
@@ -133,6 +133,20 @@ static int rt711_rtd_init(struct snd_soc_pcm_runtime *rtd)
 	return ret;
 }
 
+int sof_sdw_rt711_exit(struct device *dev)
+{
+	struct device *sdw_dev;
+
+	sdw_dev = bus_find_device_by_name(&sdw_bus_type, NULL,
+					  "sdw:0:25d:711:0");
```

Comment:
@bardliao  I wanted to refine this hard code but missed.  rt711 does not always work on link0. we can generate this name with adr info. But here not enough info is available.

Reply:
An example for you. 

````diff
+       unsigned int version, mfg, part, class;
+       char prop_name[64];
        int ret;

        /*
@@ -147,7 +149,17 @@ int sof_sdw_rt711_init(const struct snd_soc_acpi_link_adr *link,
        if (!playback)
                return 0;

-       ret = rt711_add_codec_device_props("sdw:0:25d:711:0");
+       version = SDW_VERSION(link->adr_d->adr);
+       mfg = SDW_MFG_ID(link->adr_d->adr);
+       part = SDW_PART_ID(link->adr_d->adr);
+       class = SDW_CLASS_ID(link->adr_d->adr);
+       ret = snprintf(prop_name, 64, "sdw:%x:%x:%x:%x", version,
+                mfg, part, class);
+       if (ret < 0)
+               return ret;
+
+       dev_err(NULL, " add props %s", prop_name);
+       ret = rt711_add_codec_device_props(prop_name);

````

Reply:
@RanderWang Thanks. We can get codec name from dai link.